pkg/cluster/controller.go

```diff
@@ -52,12 +53,18 @@ type ControllerConfig struct {
 }
 
 type Controller struct {
-	Config ControllerConfig
-	Log    *zap.SugaredLogger
+	Config       ControllerConfig
+	SyncExecutor k8s.SyncExecutorInterface
+	Log          *zap.SugaredLogger
 
 	sync.Mutex
 }
 
 func NewController(config ControllerConfig, log *zap.SugaredLogger) *Controller {
-	return &Controller{Config: config, Log: log}
+	syncExecutor := k8s.NewSyncExecutor(config.Client.CoreV1(), config.ClientConfig)
```

Comment:
What the NewSyncExecutor does? 
Why do we need it?

Reply:
I could not figure out how to use k8s test client with remotecommand so I instead mocked the container exec. If you can point me to an example I can remove this mock.

Reply:
I do not think fake client work within. We can check afterwords if ENV TEST allow us to to that. 
I would need to spend so effort researching and doing some POCs either. So, we can let it for future follow ups.